client/src/components/ListingForm.js

```diff
@@ -76,72 +79,99 @@ export default class ListingForm extends Component {
-              type="submit"
-            >
-              Create Listing
-            </button>
-          </div>
-        </form>
-        {this.state.showNotice && (
-          <div id="submission-notice">
-            <p>Your food listing has been successfully added.</p>
-          </div>
-        )}
-      </section>
+      <>
+        <userContext.Consumer>
+          {({ user }) => {
+            return (
+              <>
+                <section className="section w-50 m-auto">
+                  <form
+                    className="listing-form"
+                    onSubmit={this.handleSubmit.bind(this)}
+                  >
+                    <div className="form-group">
+                      <label className="mt-2 mb-2" htmlFor="meals">
+                        Number of Meals
+                      </label>
+                      <input
+                        className="form-control"
+                        type="number"
+                        name="meals"
+                        id="meals"
+                        min="1"
+                        max="1200"
+                        value={this.state.numberOfMeals}
+                        onChange={this.updateNumberOfMeals}
+                      />
+                    </div>
+
+                    <div className="form-group">
+                      <label className="mt-2 mb-2" htmlFor="desc">
+                        Description
+                      </label>
+                      <textarea
+                        className="form-control"
+                        type="text"
+                        name="desc"
+                        id="desc"
+                        rows="3"
+                        value={this.state.description}
+                        onChange={this.updateDescription}
+                      ></textarea>
+                      <small
+                        id="listingDescGuidance"
+                        className="form-text text-muted"
+                      >
+                        Please describe the type of food along with the dietary
+                        requirements.
+                      </small>
+                    </div>
+
+                    <div className="form-group">
+                      <label className="mt-2 mb-2" htmlFor="timeAvailableUntil">
+                        Available Until:
+                      </label>
+                      <input
+                        className="form-control"
+                        type="datetime-local"
+                        name="timeAvailableUntil"
+                        id="timeAvailableUntil"
+                        onChange={this.updateDateTime.bind(this)}
+                      />
+                    </div>
+
+                    <div className="form-group">
+                      <input
+                        className="form-control"
+                        type="hidden"
+                        name="listedBy"
```

Comment:
reason being - we will need to use the listed by ID to look up and get the restaurant anyway, and the easiest way to do this is using `FindById` on the model (I just wrote a `accounts/:id` get route which relies on this) to get all the info about address anyway. So since we're doing anyway that, we can easily access the username along with the address, so it makes more sense to just store the ID within the listing

Reply:
maybe I'm stuck in a very tabular way of thinking but I just believe that storing the unique ID which is generated via the DB to use as the unique look up key for another collection is better practice

Reply:
Ok I will change this back to id.